File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common/Initialization/AccessTokenProvider.fs
module AccessTokenProvider

open AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common
open System.Threading.Tasks

let getAccessTokenForUserAsync entraIDApp userName password =
    let accessToken = TestData.GetAccessTokenForUserOnMicrosoftEntraAppAsync entraIDApp
    
    Task.FromResult(accessToken userName password)

let getAccessTokenForUserOnWebClient1Async : string -> string-> Task<string> = 
        getAccessTokenForUserAsync TestData.Web1ClientApp

let getAccessTokenForUserOnWebClient2Async : string -> string-> Task<string> = 
        getAccessTokenForUserAsync TestData.Web2ClientApp

File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common.CSharp/Initialization/AccessTokenProvider.cs
using AuthZI.Identity.MicrosoftEntra;
using System;
using System.Threading.Tasks;

namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common.Initialization;

public static class AccessTokenProvider
{
  private static Task<string> GetAccessTokenForUserAsync(
    MicrosoftEntraApp entraIdApp,
    string userName)
  {
    if (!TestData.UserPasswords.TryGetValue(userName, out var password))
    {
      throw new InvalidOperationException($"Password for test user '{userName}' is not configured.");
    }

    var accessToken = TestData.GetAccessTokenForUserOnMicrosoftEntraAppAsync(entraIdApp, userName, password);

    return Task.FromResult(accessToken);
  }

  public static Task<string> GetAccessTokenForUserOnWebClient1Async(string userName) =>
    GetAccessTokenForUserAsync(TestData.Web1ClientApp, userName);

  public static Task<string> GetAccessTokenForUserOnWebClient2Async(string userName) =>
    GetAccessTokenForUserAsync(TestData.Web2ClientApp, userName);
}